cmd/apiProviderDownload_test.go

```diff
@@ -0,0 +1,91 @@
+package cmd
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+type apiProviderDownloadTestUtilsBundle struct {
+	*mock.ExecMockRunner
+	*mock.FilesMock
+}
+
+func apiProviderDownloadMockUtilsBundle() *apiProviderDownloadTestUtilsBundle {
+	utilsBundle := apiProviderDownloadTestUtilsBundle{
+		ExecMockRunner: &mock.ExecMockRunner{},
+		FilesMock:      &mock.FilesMock{},
+	}
+	return &utilsBundle
+}
+
+//Successful API Provider download cases
+func TestApiProviderDownloadSuccess(t *testing.T) {
+	t.Parallel()
+	t.Run("Successful Download of API Provider", func(t *testing.T) {
+		file, err := ioutil.TempFile("", "provider1.json")
+		if err != nil {
+			t.FailNow()
+		}
+		defer os.RemoveAll(file.Name()) // clean up
```

Comment:
Is there a reason for not mocking this?

Reply:
I have already mocked it. Refer line #43. I have removed the code with the actual ioutil calls, as it is not required.